lib/mix/tasks/compile/live_view.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Mix.Tasks.Compile.LiveView do
+  use Mix.Task
+  @recursive true
+
+  alias Mix.Task.Compiler.Diagnostic
+
+  @doc false
+  def run(args) do
+    {compile_opts, _argv, _errors} = OptionParser.parse(args, strict: [return_errors: :boolean])
+
+    diagnostics = validate_components_calls(project_modules())
+
+    if !compile_opts[:return_errors] do
+      print_diagnostics(diagnostics)
+    end
+
+    {:ok, diagnostics}
+  end
+
+  @doc false
+  def validate_components_calls(modules) do
+    for module <- modules,
+        Code.ensure_loaded?(module),
+        function_exported?(module, :__components_calls__, 0),
+        call <- module.__components_calls__() do
+      [validate_required_attrs(call), validate_undefined_attrs(call)]
+    end
+    |> List.flatten()
+  end
+
+  defp validate_required_attrs(call) do
+    %{component: {module, fun}, attrs: call_attrs, file: file, line: line} = call
+    attrs_defs = module.__components__()[fun] || []
+    passed_attrs = Enum.map(call_attrs, &elem(&1, 0))
+
+    for %{name: name, opts: opts} <- attrs_defs, opts[:required], "#{name}" not in passed_attrs do
+      message = "missing required attribute `#{name}` for component `#{inspect(module)}.#{fun}/1`"
+      error(message, file, line)
+    end
+  end
+
+  defp validate_undefined_attrs(call) do
```

Comment:
What happens if we are calling a component and the component does not actually define `attrs`? For backwards compatibility, we should not raise anything or log anything.